shared/src/main/java/com/couchbase/lite/MValueDelegate.java

```diff
@@ -44,6 +44,8 @@ public Object toNative(MValue mv, MCollection parent, AtomicBoolean cacheIt) {
             case kFLDict:
                 cacheIt.set(true);
                 return mValueToDictionary(mv, parent);
+            case kFLData:
+                return new Blob("application/octet-stream", value.asData());
```

Comment:
Need to talk about this later since this case will no longer be hit after the blob metadata fix.

Reply:
This will be fixed in a separate PR.